llvm/test/CodeGen/RISCV/pr135206.ll

```diff
@@ -0,0 +1,118 @@
+; CHECK-NEXT:    .cfi_def_cfa_offset 21328
+; CHECK-NEXT:    csrr t1, vlenb
+; CHECK-NEXT:    .cfi_def_cfa t1, -8
+; CHECK-NEXT:    lui t2, 1
+; CHECK-NEXT:  .LBB0_3: # =>This Inner Loop Header: Depth=1
+; CHECK-NEXT:    sub sp, sp, t2
+; CHECK-NEXT:    sd zero, 0(sp)
+; CHECK-NEXT:    sub t1, t1, t2
+; CHECK-NEXT:    bge t1, t2, .LBB0_3
+; CHECK-NEXT:  # %bb.4:
+; CHECK-NEXT:    .cfi_def_cfa_register sp
+; CHECK-NEXT:    sub sp, sp, t1
+; CHECK-NEXT:    .cfi_escape 0x0f, 0x0f, 0x72, 0x00, 0x11, 0xc0, 0xb6, 0x01, 0x22, 0x11, 0x01, 0x92, 0xa2, 0x38, 0x00, 0x1e, 0x22 # sp + 23360 + 1 * vlenb
+; CHECK-NEXT:    li s0, 0
+; CHECK-NEXT:    li a0, 170
+; CHECK-NEXT:    li s1, 32
+; CHECK-NEXT:    lui a1, %hi(.LCPI0_0)
+; CHECK-NEXT:    vsetivli zero, 16, e8, m1, ta, ma
+; CHECK-NEXT:    vmv.v.x v8, a0
+; CHECK-NEXT:    lui a0, 6
+; CHECK-NEXT:    addiw a0, a0, -1264
+; CHECK-NEXT:    add a0, sp, a0
+; CHECK-NEXT:    vs1r.v v8, (a0) # vscale x 8-byte Folded Spill
+; CHECK-NEXT:    ld s2, %lo(.LCPI0_0)(a1)
+; CHECK-NEXT:    vse8.v v8, (s1)
+; CHECK-NEXT:    li s3, 16
+; CHECK-NEXT:    vse8.v v8, (s3)
+; CHECK-NEXT:    vse8.v v8, (zero)
+; CHECK-NEXT:    sd s2, 48(zero)
+; CHECK-NEXT:    jalr s0
+; CHECK-NEXT:    lui a0, 6
+; CHECK-NEXT:    addiw a0, a0, -1264
+; CHECK-NEXT:    add a0, sp, a0
+; CHECK-NEXT:    vl1r.v v8, (a0) # vscale x 8-byte Folded Reload
+; CHECK-NEXT:    vsetivli zero, 16, e8, m1, ta, ma
+; CHECK-NEXT:    vse8.v v8, (zero)
+; CHECK-NEXT:    vse8.v v8, (s3)
+; CHECK-NEXT:    vse8.v v8, (s1)
+; CHECK-NEXT:    sd s2, 48(zero)
+; CHECK-NEXT:    li a0, 0
+; CHECK-NEXT:    csrr a1, vlenb
+; CHECK-NEXT:    add sp, sp, a1
+; CHECK-NEXT:    .cfi_def_cfa sp, 2032
+; CHECK-NEXT:    lui a1, 5
+; CHECK-NEXT:    addiw a1, a1, 848
+; CHECK-NEXT:    add sp, sp, a1
+; CHECK-NEXT:    .cfi_def_cfa_offset 2032
+; CHECK-NEXT:    ld ra, 2024(sp) # 8-byte Folded Reload
+; CHECK-NEXT:    ld s0, 2016(sp) # 8-byte Folded Reload
+; CHECK-NEXT:    ld s1, 2008(sp) # 8-byte Folded Reload
+; CHECK-NEXT:    ld s2, 2000(sp) # 8-byte Folded Reload
+; CHECK-NEXT:    ld s3, 1992(sp) # 8-byte Folded Reload
+; CHECK-NEXT:    .cfi_restore ra
+; CHECK-NEXT:    .cfi_restore s0
+; CHECK-NEXT:    .cfi_restore s1
+; CHECK-NEXT:    .cfi_restore s2
+; CHECK-NEXT:    .cfi_restore s3
+; CHECK-NEXT:    addi sp, sp, 2032
+; CHECK-NEXT:    .cfi_def_cfa_offset 0
+; CHECK-NEXT:    ret
+  %1 = alloca %"struct.fuchsia_sysmem::wire::BufferCollectionConstraints", align 8
+  %2 = alloca %"struct.fuchsia_sysmem::wire::BufferCollectionConstraints", align 8
+  %3 = alloca %"struct.fuchsia_sysmem::wire::BufferCollectionConstraints", align 8
+  call void @llvm.memset.p0.i64(ptr null, i8 -86, i64 56, i1 false)
+  %4 = call ptr null()
+  call void @llvm.memset.p0.i64(ptr null, i8 -86, i64 56, i1 false)
+  ret i1 false
+}
+
+attributes #0 = { nocallback nofree nounwind willreturn memory(argmem: write) }
+attributes #1 = { "probe-stack"="inline-asm" "target-features"="+v" }
```

Comment:
Is V important for the reproducer?

Reply:
not sure. Let me see if it still works w/o it.

Reply:
Yeah, `+v` did seem to be required. Looking at the fix again, that makes sense due to the `(MI->getOpcode() == RISCV::PROBED_STACKALLOC_RVV)` bits. I'm guessing that if only one of the stack probes gets used, it wouldn't foul up the iterator.